### PR TITLE
py-aipy: a new package. [WIP]

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-aipy/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    url      = "https://github.com/HERA-Team/aipy/archive/v3.0.1.tar.gz"
+
+    version('3.0.1', sha256='27b61f565ce135812fef4f9fed5b9ec88f6aae635e326b21613ce675f876f16c')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-basemap', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-astropy', type=('build', 'run'))
+    depends_on('py-matplotlib@:2.9.9', type=('build', 'run'), when='^python@2.7:2.9')
+    depends_on('py-matplotlib', type=('build', 'run'), when='^python@3.4:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-aipy/package.py:23]: These constraints are actually due to matplotlib, not due to aipy. I would use:
```python
depends_on('py-matplotlib', type=('build', 'run'))
```
and let matplotlib sort things out.

File: var/spack/repos/builtin/packages/py-aipy/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    homepage = "https://github.com/HERA-Team/aipy"
+    url      = "https://github.com/HERA-Team/aipy/archive/v3.0.1.tar.gz"
+
+    version('3.0.1', sha256='27b61f565ce135812fef4f9fed5b9ec88f6aae635e326b21613ce675f876f16c')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-basemap', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-astropy', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-aipy/package.py:21]:
```suggestion
    depends_on('py-astropy@1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-aipy/package.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    collects together tools for radio astronomical interferometry."""
+
+    homepage = "https://github.com/HERA-Team/aipy"
+    url      = "https://github.com/HERA-Team/aipy/archive/v3.0.1.tar.gz"
+
+    version('3.0.1', sha256='27b61f565ce135812fef4f9fed5b9ec88f6aae635e326b21613ce675f876f16c')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-basemap', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-aipy/package.py:20]:
```suggestion
    depends_on('py-numpy@1.2:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-aipy/package.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on('py-astropy', type=('build', 'run'))
+    depends_on('py-matplotlib@:2.9.9', type=('build', 'run'), when='^python@2.7:2.9')
+    depends_on('py-matplotlib', type=('build', 'run'), when='^python@3.4:')
+    depends_on('py-ephem@3.7.3.2:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-aipy/package.py:24]: Missing some dependencies:
```python
    install_requires = [                                                        
        ASTROPY_DEP,                                                            
        'healpy>=1.11',                                                         
        MATPLOTLIB_DEP,                                                         
        'numpy>=1.2',                                                           
        'ephem>=3.7.3.2',                                                       
        'scipy>=0.19',                                                          
    ],   
```

Comment (weijianwen) [reply var/spack/repos/builtin/packages/py-aipy/package.py:24]: `py-healpy` is on the way.

**Comment by adamjstewart** [comment]
It looks like `py-healpy` has been merged, can you update this PR?

**Comment by alalazo** [comment]
Closing as stale. Feel free to reopen if you want to continue working on this.

**Pull request closed without merging**